src/main/java/com/sapred/ordermanagerred/service/OrderService.java

```diff
@@ -54,6 +66,50 @@ public List<Order> getOrders(String token, String statusId, int pageNumber, Stri
         return pageOrders.getContent();
     }
 
+    public List<Order> getOrdersByFilters(Map<String, Object> filterMap, String token, int pageNumber) {
+        if (!filterMap.containsKey("companyId")){
+            throw new MapFilterMissedField("the companyId key is require!");
+        }
+        MongoCollection<Document> orderCollection = mongoTemplate.getCollection("Order");
+
+        List<Bson> filters = new ArrayList<>();
+        // Iterate through the filter map and construct filters for each entry
+        for (Map.Entry<String, Object> entry : filterMap.entrySet()) {
+            String filterName = entry.getKey();
+            Object filterValue = entry.getValue();
+
+            filters.add(Filters.eq(filterName, filterValue));
+        }
+        Bson finalFilter = Filters.and(filters);
```

Comment:
I can't see any usage of finalFilter
what I missed?

Reply:
Try to run 
FindIterable<Order> ordersDucuments = orderCollection.find(finalFilter, Order.class);
I think in this case the result will be Order and not Document so no need converter


Update me if not work I will try to find the conversion in more nice way

Reply:
sorry, my mistake. I had an error after the merging so I put the relavant line in comment fin order to check just the base case and forgot take it out...
now this idea  orderCollection.find(finalFilter, Order.class) was realy  great. It works a little bit different so I changed it...